docs/MIDDLEWARES.md

```diff
@@ -0,0 +1,85 @@
+# Internal libraries
+This is a set of internal utilities being used by the `Hull.App` and `Handlers`.
+They should be used in case of special requirements or a need of greater customization.
+
+## Hull.Middleware()
+This middleware standardizes the instantiation of a Hull client from configuration passed as a Query string or as a token. It also optionally fetches the entire ship's configuration and caches it to save requests.
+
+### Params
+
+##### hostSecret
+The ship hosted secret (Not the one received from Hull. The one the hosted app itself defines. Will be used to encode tokens).
+
+##### clientConfig
+Additional config which will be passed to the new instance of Hull Client
+
+```js
+import Hull from "hull";
+
+app.use(Hull.Middleware({ hostSecret: "supersecret" }));
+
+app.use((req, res) => { res.json({ message: "thanks" }); });
+
+app.use(function(err, res, req, next){
```

Comment:
same here. this MUST be something the developer can do. a generic error catching middleware to avoid crashes where he can decide what to return

Cant' take this away from them